ios/browser/api/ads/brave_ads.mm

```diff
@@ -241,6 +246,13 @@ - (void)initializeWithSysInfo:(BraveAdsSysInfo*)sysInfo
              buildChannelInfo:(BraveAdsBuildChannelInfo*)buildChannelInfo
                    walletInfo:(BraveAdsWalletInfo*)walletInfo
                    completion:(void (^)(bool))completion {
+  if ([self isAdsServiceRunning]) {
+    return;
```

Comment:
We should _probably_ call `completion` here before returning, but not sure if we want to always pass in `true` since im not sure its guaranteed that initialization has completed if we were to call it a second time while initializing (ads ptr would be non-nil, but ads may not be fully initialized yet). Though I don't think we call initialize more than once on the iOS side anyways

Reply:
If initialized previously failed, then the only way initialization is likely to succeed is if the browser is upgraded to fix the cause of failure. But I am for always returning something for the `completion`

Reply:
I've set it to return `true` for this case